src/aria/utils/Data.js

```diff
@@ -366,24 +366,45 @@ module.exports = Aria.classDefinition({
          * @param {Object} messages
          * @param {Array} groups an optional parameter that contains the group(s) to be validated, if no group is
          * specified then validation will occur on the model as normal.
-         * @param {Boolean} stopOnError an optional parameter that (when set to true) stops the validation as soon as an error is detected in
-         * the model provided (defaults to false)
+         * @param {Boolean} stopOnError an optional parameter that (when set to true) stops the validation as soon as an
+         * error is detected in the model provided (defaults to false)
+         * @param {Object} Map stores a reference to an already validated part of the model
          */
         validateModel : function (dataHolder, messages, groups, stopOnError) {
-            if (this.utilsType.isObject(dataHolder)) {
-                for (var key in dataHolder) {
-                    if (dataHolder.hasOwnProperty(key)) {
-                        this.__subValidateModel(dataHolder, key, messages, groups, stopOnError);
-                        if (stopOnError && messages && messages[this.errorNbrKey] > 0){
-                            return;
+            var map = new ariaUtilsStackHashMap();
+            this._validateModel(dataHolder, messages, groups, stopOnError, map);
+            map.$dispose();
+        },
+
+        /**
+         * validates an object within the data model, including all fields values and validation stacks for those fields
+         * @param {Object} dataHolder
+         * @param {Object} messages
+         * @param {Array} groups an optional parameter that contains the group(s) to be validated, if no group is
+         * specified then validation will occur on the model as normal.
+         * @param {Boolean} stopOnError an optional parameter that (when set to true) stops the validation as soon as an
+         * error is detected in the model provided (defaults to false)
+         * @param {Object} Map stores a reference to an already validated part of the model
+         * @protected
```

Comment:
There is no need to have the `@protected` keyword here, as this is automatically deduced from the underscore in the name of the method.
The type of the map parameter is `aria.utils.StackHashMap`.